### PR TITLE
Load Balancer WAF control BRS + documentation

### PR DESCRIPTION
**Pull request by HarshitArora8**

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1 [hunk 1]
```diff
@@ -0,0 +1,825 @@
+<###
+# Overview:
+    This script is used to configure the NSG on Subnet being used in the Load Balancer in a Subscription.
+# Control ID:
+    Azure_LoadBalancer_NetSec_Enable_WAF_Trial
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1]: Fixed

File: Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1 [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+        3. Execute the script to configure NSG on the Subnet(s) in the Subscription. Refer `Examples`, below.
+    To roll back:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to Remove the NSG configuration on the Subnet(s) in the Subscription. Refer `Examples`, below.
+# Examples:
+    To remediate:
+        1. To review the Subnet(s) in a Subscription that will be remediated:
+    
+           File has already been generated using the previous script.
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1]: Removed.

File: Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1 [hunk 3]
```diff
@@ -0,0 +1,825 @@
...
+        .PARAMETER DryRun	
+        Specifies a dry run of the actual remediation.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+        
+        .PARAMETER FilePath
+        Specifies the path to the file to be used as input for the remediation.
+        .INPUTS
+        None. You cannot pipe objects to Add-NSGConfigurationOnSubnet.
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1:123]: Added.

File: Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1 [hunk 4]
```diff
@@ -0,0 +1,825 @@
...
+        1. To review the Subnet(s) in a Subscription that will be remediated:
+    
+           File has already been generated using the previous script.
+        2. Configure the NSG on the Subnet(s)(s) in the Subscription:
+       
+           Add-NSGConfigurationOnSubnet -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+        3. Configure the NSG on the Subnet(s) in the Subscription, from a previously taken snapshot:
+       
+           Add-NSGConfigurationOnSubnet -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\ConfigureNSG\SubnetDetailsBackUp.csv
+        To know more about the options supported by the remediation command, execute:
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1:47]: Fixed,

File: Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1 [hunk 5]
```diff
@@ -0,0 +1,825 @@
...
+
+function Add-NSGConfigurationOnSubnet
+{
+    <#
+        .SYNOPSIS
+        Remediates 'Azure_LoadBalancer_NetSec_Enable_WAF_Trial' Control.
+
+        .DESCRIPTION
+        Remediates 'Azure_LoadBalancer_NetSec_Enable_WAF_Trial' Control.
+        Add the NSG configuration on the Subnet(s) in the Subscription.
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1]: Corrected.

File: Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1 [hunk 6]
```diff
@@ -0,0 +1,822 @@
...
+        }
+
+        Write-Host "Fetch all Subnet(s) from [$($FilePath)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $LoadBalancerSubnetResources = Import-Csv -LiteralPath $FilePath
+
+        $validLoadBalancerSubnetResources = $LoadBalancerSubnetResources| Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+
+
+        $validLoadBalancerSubnetResources| ForEach-Object{
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Scripts/RemediationScripts/Remediate-NSGConfigurationOnLoadBalancerSubnet.ps1:287]: Fixed.

File: Control coverage/Feature/LoadBalancer.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+- **PowerShell** 
+	# Below commands will be useful to Configure WAF on Load Balancer
+    Connect-AzAccount
+	Set-AzContext -SubscriptionId "<sub id>"
+	# Get Load Balancer
+	Get-AzLoadBalancer -ResourceGroupName "<ResourceGroup>" -Name "<ResourceName>"
+	
+	# Get Virtual Network and its Subnets
+	Get-AzVirtualNetwork -Name "<ResourceVNName>" -ResourceGroupName "<ResourceVNRGName>"
+    Get-AzVirtualNetworkSubnetConfig -VirtualNetwork "<VirtualNetwork>" -Name "<SubNetName>"
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Control coverage/Feature/LoadBalancer.md]: Fixed.

File: Control coverage/Feature/LoadBalancer.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+WAF enforcement on the Load Balancer further strengthens the security posture of your applications by protecting them from the common web vulnerabilities. This allows you to secure both your internet-facing as well as your internal application workloads.
+
+### Control Spec 
+
+> **Passed:** 
+> 1. Web Application Firewall has been configured on Load Balancer. <b>OR</b>
+> 2. Network Security Group is configured on the subnets associated with the Load Balancer.
+> 
+> **Failed:** 
+> 1. Web Application Firewall is not configured on Load Balancer. <b>OR</b>
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Control coverage/Feature/LoadBalancer.md]: Fixed

File: Control coverage/Feature/LoadBalancer.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+> **Error:** 
+>There was an error fetching WAF Configuration details of Load Balancer.
+ 
+### Recommendation
+- **Azure Portal** 
+
+	 To Remediate WAF on Load balancer attach every subnet of virtual network with NSG/Azure Firewall.
+
+- **PowerShell** 
+	# Below commands will be useful to Configure WAF on Load Balancer
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Control coverage/Feature/LoadBalancer.md]: Fixed.

File: Control coverage/Feature/LoadBalancer.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+### Display Name 
+[Trial] Load Balancer should have Web Application Firewall (WAF)
+
+### Rationale 
+WAF enforcement on the Load Balancer further strengthens the security posture of your applications by protecting them from the common web vulnerabilities. This allows you to secure both your internet-facing as well as your internal application workloads.
+
+### Control Spec 
+
+> **Passed:** 
+> 1. Web Application Firewall has been configured on Load Balancer. <b>OR</b>
```
[the remark this answers is not in the document]
Comment (HarshitArora8) [reply Control coverage/Feature/LoadBalancer.md]: Fixed.

**Review by jaiswalricha — APPROVED** [review]

**Review by VamseeInala — APPROVED** [review]
Call out that only connecting to NSG is part of this remediation script and not connecting to Azure Firewall

**Pull request merged**